<launch>
  <arg name="start_nodelet_manager" default="true"/>
  <arg name="nodelet_manager_name" default="nodelet_manager_control"/>
  <arg name="mass"/>
  <arg name="odom_topic" default="odom"/>
  <arg name="config_path"/>
  <arg name="thrust_config_file"/>
  <arg name="gains_config_file"/>

  <node pkg="nodelet"
    type="nodelet"
    name="$(arg nodelet_manager_name)"
    args="manager"
    output="screen"
    if="$(arg start_nodelet_manager)"/>

  <node pkg="nodelet"
    type="nodelet"
    name="trackers_manager"
    args="load kr_trackers_manager/TrackersManager $(arg nodelet_manager_name)"
    output="screen">
    <rosparam file="$(find kr_mav_launch)/config/trackers.yaml"/>
    <rosparam file="$(find kr_mav_launch)/config/tracker_params.yaml"/>
    <rosparam file="$(arg config_path)/$(arg gains_config_file)"/>
    <remap from="~odom" to="$(arg odom_topic)"/>
    <remap from="~cmd" to="position_cmd"/>
  </node>

  <node pkg="nodelet"
    type="nodelet"
    args="load kr_mav_controllers/SO3ControlNodelet $(arg nodelet_manager_name)"
    name="so3_control"
    required="true"
    output="screen">
    <rosparam file="$(arg config_path)/$(arg gains_config_file)"/>
    <param name="mass" value="$(arg mass)"/>
    <remap from="~odom" to="$(arg odom_topic)"/>
    <remap from="~position_cmd" to="position_cmd"/>
    <remap from="~so3_cmd" to="so3_cmd"/>
    <remap from="~motors" to="motors"/>
  </node>

  <!-- MAV Manager -->
  <node pkg="kr_mav_manager"
    type="mav_services"
    name="mav_services"
    output="screen">
    <rosparam file="$(arg config_path)/mav_manager_params.yaml"/>
    <remap from="odom" to="$(arg odom_topic)"/>
  </node>

</launch>